src/QsCompiler/BondSchemas/Protocols.cs

```diff
@@ -39,34 +66,52 @@ public static class Protocols
             SyntaxTree.QsCompilation qsCompilation,
             Stream stream)
         {
-            var outputBuffer = new OutputBuffer();
-            var serializer = GetSimpleBinarySerializer();
-            var writer = new SimpleBinaryWriter<OutputBuffer>(outputBuffer);
-            var bondCompilation = BondSchemaTranslator.CreateBondCompilation(qsCompilation);
-            serializer.Serialize(bondCompilation, writer);
-            stream.Write(outputBuffer.Data);
+            lock(BondSharedDataStructuresLock)
+            {
+                var outputBuffer = new OutputBuffer();
+                var serializer = GetSimpleBinarySerializer();
+                var writer = new SimpleBinaryWriter<OutputBuffer>(outputBuffer);
+                var bondCompilation = BondSchemaTranslator.CreateBondCompilation(qsCompilation);
+                serializer.Serialize(bondCompilation, writer);
+                stream.Write(outputBuffer.Data);
+            }
+
             stream.Flush();
             stream.Position = 0;
         }
 
-        private static Deserializer<SimpleBinaryReader<InputBuffer>> GetSimpleBinaryDeserializer()
+        private static SimpleBinaryDeserializer GetSimpleBinaryDeserializer()
         {
-            if (simpleBinaryDeserializer == null)
+            Debug.Assert(Monitor.IsEntered(BondSharedDataStructuresLock));
```

Comment:
Debug.Asserts are not properly registering when running the extensions in debug and there were at some point also certain cross-platform issues (not sure if there still are). As awkward as it is, could we instead use 
     #if DEBUG
     // a check that throws if it is not satisfied
     #endif
Same for the other Debug.Asserts.

Reply:
Replaced Debug.Assert with the `VerifyLockAcquired` that uses `#if DEBUG`.